lib/puppet_x/puppetlabs/kubernetes/provider.rb

```diff
@@ -103,7 +103,7 @@ def make_object(type, name, params)
           params[:metadata] = {} unless params.key?(:metadata)
           p = params.swagger_symbolize_keys
           object = Object::const_get("Kubeclient::#{klass}").new(p)
-          object.metadata.name = name
+          object.metadata.name = kubename unless kubename.nil?
```

Comment:
if `kubename` is nil, then the object will be initialized without a name at all. This is possible for a `kubernetes_namespace` resource I think.
You might want to do something like this maybe:
~~~ruby
object.metadata.name = kubename || name
~~~
So that if there's no `metadata['name']` then we use the Puppet resource name to init it.

Reply:
Yes I agree. I do think it is better to use the metadata/name thought. But if we forget about it